### ISSUE TITLE
Support for Rust-written programs

### ISSUE THREAD

**Issue by Aandreba** [Aandreba/blaze#16]
Studying adding support for Rust-written programs.
The current plan would be to create a program that would facilitate the integration of Blaze with [rust-gpu](https://github.com/EmbarkStudios/rust-gpu). Further details to be specified.

**Comment by hughesjs** [comment]
Is this in an effort to avoid needing to embed a C program into the application like how the official `opencl3` crate recommends?

![image](https://user-images.githubusercontent.com/4203766/222874526-2cde809c-db5c-424a-a52c-3cc9c941236e.png)

**Comment by Aandreba** [comment]
Kind of. If you want to avoid writing C code directly inside a string, you can use the standard library's `include_str!` macro to add the contents of a file directly into a string literal.
Apart from that, the general idea is to make it easier to include shaders written with Rust & rust-gpu.

**Comment by hughesjs** [comment]
Yeah, I was asking because it would be really nice to be able to write the actual compute code in Rust and transpile it to OpenCL C

**Comment by Aandreba** [comment]
You can write shaders with the **rust-gpu** project and then use [`RawProgram::from_il`](https://docs.rs/blaze-rs/latest/blaze_rs/core/struct.RawProgram.html#method.from_il) to import the generated spv file, but support for importing spv files with the `blaze` macro is still yet to be implemented, unfortunately.
